### PR TITLE
CLOUD-371 namespace selection

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-51-5ba185f

**Review by cap1984 — APPROVED** [review]

File: broker/cluster.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ func (p *Controller) DeployCluster(instance ServiceInstance, skipS3Storage *bool
 	if err != nil {
 		return err
 	}
+	if len(instance.Context.Namespace) > 0 {
+		dbservice.Namespace = instance.Context.Namespace
+	}
```
Comment (dAdAbird) [inline broker/cluster.go]: No need in check. `dbservice.Namespace` already is an empty string. So if `instance.Context.Namespace == ""`  `dbservice.Namespace` remains the same.

File: dbaas/dbaas.go [hunk 2]
```diff
@@ -111,11 +112,18 @@ func (p Cmd) readOperatorLogs(operatorName string) ([]byte, error) {
 }
 
 func (p Cmd) GetObject(typ, name string) ([]byte, error) {
+	if len(p.Namespace) > 0 {
+		return p.runCmd(execCommand, "get", typ+"/"+name, "-n", p.Namespace, "-o", "json")
+	}
```
Comment (dAdAbird) [inline dbaas/dbaas.go:117]: Maybe it's better to make the namespace selection the part of the `runCmd` method?

File: dbaas/delete.go [hunk 3]
```diff
@@ -36,9 +36,16 @@ func (p Cmd) Delete(typ string, app Deploy, delPVC bool, ok chan<- string, errc
...
+	if len(p.Namespace) > 0 {
+		out, err := p.runCmd("kubectl", "delete", typ, name, "-n", p.Namespace)
+		if err != nil {
+			return errors.Wrapf(err, "get cr: %s", out)
+		}
+	} else {
+		out, err := p.runCmd("kubectl", "delete", typ, name)
+		if err != nil {
+			return errors.Wrapf(err, "get cr: %s", out)
+		}
```
Comment (dAdAbird) [inline dbaas/delete.go]:
```go
var args []string
if len(p.Namespace) > 0 {
    args = []string{"-n", p.Namespace}
}
out, err := p.runCmd("kubectl", "delete", typ, name, args...)
return errors.Wrapf(err, "get cr: %s", out)
```

File: broker/cluster.go [hunk 4]
```diff
@@ -189,6 +194,13 @@ func (p *Controller) UpdateCluster(instance *ServiceInstance) error {
 	created := make(chan string)
 	msg := make(chan dbaas.OutuputMsg)
 	cerr := make(chan error)
+	dbservice, err = dbaas.New(p.EnvName)
```
Comment (delgod) [inline broker/cluster.go:197]: maybe we can pass names pace identical to env? is it have sense?

**Review by dAdAbird — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-51-e9e5ff3

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:master-broker-PR-51-7eb7ea3

File: dbaas/dbaas.go [hunk 5]
```diff
@@ -40,10 +40,12 @@ const (
 	PlatformMinishift  PlatformType = "minishift"
 )
 
-var execCommand string
+//var execCommand string
```
Comment (dAdAbird) [inline dbaas/dbaas.go]: remove pls

File: dbaas/dbaas.go [hunk 6]
```diff
@@ -57,7 +59,7 @@ func (e ErrCmdRun) Error() string {
 }
 
 func New(environment string) (*Cmd, error) {
-	execCommand = k8sExecDefault
+	execCommand := k8sExecDefault
 	if _, err := exec.LookPath(execCommand); err != nil {
 		execCommand = k8sExecCustom
 		if _, err := exec.LookPath(execCommand); err != nil {
```
Comment (dAdAbird) [inline dbaas/dbaas.go:63]: no need in panic below. just return the error

**Review by dAdAbird — CHANGES_REQUESTED** [review]
don't panic, see comments

File: dbaas/dbaas.go [hunk 7]
```diff
@@ -57,11 +57,11 @@ func (e ErrCmdRun) Error() string {
...
-			panic(fmt.Sprintf("Unable to find neither '%s' nor '%s' exec files", k8sExecDefault, k8sExecCustom))
+			return nil, fmt.Errorf("Unable to find neither '%s' nor '%s' exec files", k8sExecDefault, k8sExecCustom)
```
Comment (dAdAbird) [inline dbaas/dbaas.go]:
```suggestion
			return nil, fmt.Errorf("unable to find neither '%s' nor '%s' exec files", k8sExecDefault, k8sExecCustom)
```

**Review by dAdAbird — APPROVED** [review]

**Pull request merged**